gophers/chuck_norris_gopher.go

```diff
@@ -11,13 +18,35 @@ func NewChuckNorrisGopher() ChuckNorrisGopher {
 	}
 }
 
+type ChuckNorrisAPIResponse struct {
+  Response string
+  Value api_data
+}
+
+type api_data struct {
+  Id int
+  Joke string
+}
+
 func (g ChuckNorrisGopher) TransformMessage(msg string) string {
-	// TODO: Lookup quotes from an HTTP API and return them
-	// Feel free to use a different quote API if Chuck isn't to your liking
+  response, err := http.Get(g.endpoint)
+  if err != nil {
+    panic(err.Error())
+  }
+
+  // Need to take the body of the response and turn into a byte array
+  body, err := ioutil.ReadAll(response.Body)
```

Comment:
You can combine reading the bytes with unmarshaling, saves you a bit of extra error handling.

```golang
err = json.NewDecoder(resp.Body).Decode(&api_struct)
```